Expensify/App
# Refactor the API for saving the report notification preferences

- [x] Held on https://github.com/Expensify/Web-Expensify/pull/34059
- [x] Held on https://github.com/Expensify/Web-Expensify/pull/34076

### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/211600

### Tests
1. Open an #announce workspace chate
2. Click on the chat header to go to the details
3. Go to the settings
4. Change the notification preferences
5. Refresh/reload the app to ensure the setting stuck


### QA Steps
Same as above


### Screenshots


#### Web

https://user-images.githubusercontent.com/1228807/174681699-c518d970-54ed-445c-b01a-33459c78de88.mp4

Adding in a video to show pusher working to update multiple connected clients


https://user-images.githubusercontent.com/1228807/175089042-a5f795b2-938b-41c6-9728-9b8f20ada67f.mp4





#### Mobile Web

https://user-images.githubusercontent.com/1228807/174681716-c87dbf18-4a5a-40b5-96da-1b9f2ba5807e.mp4



#### Desktop

https://user-images.githubusercontent.com/1228807/174681731-22699d08-2120-4218-a23d-d1213830c1ab.mp4



#### iOS

https://user-images.githubusercontent.com/1228807/174681761-940f600a-20ab-4612-ba72-85e9ac40f004.mp4



#### Android

https://user-images.githubusercontent.com/1228807/174681741-9bbe6fef-379c-4086-911e-2e9933f5ca45.mp4
